eth/protocols/eth/handlers.go

```diff
@@ -274,6 +285,57 @@ func ServiceGetReceiptsQuery(chain *core.BlockChain, query GetReceiptsRequest) [
 	return receipts
 }
 
+// serviceGetReceiptsQuery69 assembles the response to a receipt query.
+// It does not send the bloom filters for the receipts
+func serviceGetReceiptsQuery69(chain *core.BlockChain, query GetReceiptsRequest) []rlp.RawValue {
+	// Gather state data until the fetch or network limits is reached
+	var (
+		bytes    int
+		receipts []rlp.RawValue
+	)
+	for lookups, hash := range query {
+		if bytes >= softResponseLimit || len(receipts) >= maxReceiptsServe ||
+			lookups >= 2*maxReceiptsServe {
+			break
+		}
+		// Retrieve the requested block's receipts
+		results := chain.GetRawReceiptsByHash(hash)
+		if results == nil {
+			if header := chain.GetHeaderByHash(hash); header == nil || header.ReceiptHash != types.EmptyRootHash {
+				continue
+			}
+		} else {
+			block := chain.GetBlockByHash(hash)
+			results = transformReceipts(results, block.Transactions())
+		}
+		receipts = append(receipts, results)
+		bytes += len(results)
+	}
+	return receipts
+}
+
+func transformReceipts(blockReceipts []byte, txs []*types.Transaction) []byte {
```

Comment:
please add a docstring which details what this transformation does.  Maybe something like
```suggestion
// transformReceipts takes a slice of rlp-encoded recepts, and transactions, and applies the type-encoding
// on the receipts (for non-legacy receipts). 
func transformReceipts(blockReceipts []byte, txs []*types.Transaction) []byte {
```